app/src/main/java/org/commcare/dalvik/reminders/MainActivity.kt

```diff
@@ -38,16 +39,35 @@ class MainActivity : AppCompatActivity() {
                 result.data?.let { intent ->
                     intent.getStringExtra("isPermissionGranted")?.let {permissionResult ->
                         if (permissionResult == PermissionActivity.GRANTED) {
-                            getStarted()
+                            if (hasNotificationPermission()) {
+                                getStarted()
+                            } else {
+                                requestPermissionBtn.visibility = VISIBLE
+                                var msg =  resources.getString(R.string.following_permission_not_granted)
+                                msg = msg.plus(getString(R.string.notification_permission_not_granted))
+                                statusTv.text = msg
+                                statusTv.visibility = VISIBLE
```

Comment:
can abstract in another method and resue here and below.

Reply:
done